Test/IntegrationTests/MqttClient/MqttClient/ContentView.swift

```diff
@@ -0,0 +1,225 @@
+// Update the host and port before run the app.
+let TEST_HOST = "<endpoint>"
+let TEST_PORT: UInt32 = 1883
+var mqttTestContext = MqttTestContext()
+var client: Mqtt5Client?
+
+@available(iOS 14.0, *)
+struct ContentView: View {
+    @StateObject var testContext = mqttTestContext
+    var body: some View {
+        VStack {
+            Button("Setup Client and Start") {
+                do {
+                    library_init()
+                    setupClientAndStart()
+                }
+            }
+            NavigationView {
+                List(testContext.messages) { message in
+                    HStack {
+                        Text(message.text)
+                    }
+                 }.navigationBarTitle(Text("Messages"))
+            }
+        }
+        .padding()
+    }
+
+}
+
+/// Message struct for information print
+struct Message: Identifiable {
+    let id: Int
+    let text: String
+}
+
+/// start client and check for connection success
+func connectClient(client: Mqtt5Client, testContext: MqttTestContext) throws {
+    try client.start()
+    if testContext.semaphoreConnectionSuccess.wait(timeout: .now() + 5) == .timedOut {
+        print("Connection Success Timed out after 5 seconds")
+    }
+}
+
+/// stop client and check for stopped lifecycle event
+func stopClient(client: Mqtt5Client, testContext: MqttTestContext) throws {
+    try client.stop()
+    if testContext.semaphoreStopped.wait(timeout: .now() + 5) == .timedOut {
+        print("Stop timed out after 5 seconds")
+    }
+}
+
+/// generate a random client id
+func createClientId() -> String {
+    return "aws-crt-swift-iOS-test-" + UUID().uuidString
+}
+
+class MqttTestContext: ObservableObject {
+    @Published var messages: [Message] = [Message(id: 0, text: "Click the \"Setup Client and Start\" to start the client.")]
+
+    public var contextName: String
+
+    public var onPublishReceived: OnPublishReceived?
+    public var onLifecycleEventStopped: OnLifecycleEventStopped?
+    public var onLifecycleEventAttemptingConnect: OnLifecycleEventAttemptingConnect?
+    public var onLifecycleEventConnectionSuccess: OnLifecycleEventConnectionSuccess?
+    public var onLifecycleEventConnectionFailure: OnLifecycleEventConnectionFailure?
+    public var onLifecycleEventDisconnection: OnLifecycleEventDisconnection?
+    public var onWebSocketHandshake: OnWebSocketHandshakeIntercept?
+
+    public let semaphorePublishReceived: DispatchSemaphore
+    public let semaphoreConnectionSuccess: DispatchSemaphore
+    public let semaphoreConnectionFailure: DispatchSemaphore
+    public let semaphoreDisconnection: DispatchSemaphore
+    public let semaphoreStopped: DispatchSemaphore
+
+    public var lifecycleConnectionFailureData: LifecycleConnectionFailureData?
+    public var lifecycleDisconnectionData: LifecycleDisconnectData?
+    public var publishCount = 0
+
+    /// Print the text and pending new message to message list
+    func printView(_ txt: String) {
```

Comment:
These UI updates need to happen on the main DispatchQueue. Wrap below in:
```
DispatchQueue.main.async {
let newMessage = Message(id: self.messages.count, text: txt)
self.messages.append(newMessage)
print(txt)
}
```
to resolve.